services/beaconchain/src/main/java/tech/pegasys/teku/services/beaconchain/BeaconChainController.java

```diff
@@ -853,12 +853,8 @@ protected void initDasCustody() {
             },
             metricsSystem,
             timeProvider);
+    this.dataColumnSidecarCustodySupplier = () -> dataColumnSidecarRecoveringCustody;
```

Comment:
So you change the object this var was linked to. If you want to go this way, you need to keep wrapper object and change the object wrapped. For example AtomicReference will work for this or List. Of course, using list for this would be weird, but let's try with AtomicReference to see how it will look like

Reply:
As suggested by @tbenr using Supplier and a method reference works too but imo using AtomicReference is more suitable in this case since the goal is simply to eplace the NOOP by a real implementation